sklearn/utils/tests/test_encode.py

```diff
@@ -0,0 +1,61 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from sklearn.utils._encode import _unique
+from sklearn.utils._encode import _encode
+from sklearn.utils._encode import _encode_check_unknown
+
+
+@pytest.mark.parametrize(
+        "values, expected",
+        [(np.array([2, 1, 3, 1, 3], dtype='int64'),
+          np.array([1, 2, 3], dtype='int64')),
+         (np.array(['b', 'a', 'c', 'a', 'c'], dtype=object),
+          np.array(['a', 'b', 'c'], dtype=object)),
+         (np.array(['b', 'a', 'c', 'a', 'c']),
+          np.array(['a', 'b', 'c']))],
+        ids=['int64', 'object', 'str'])
+def test_encode_util(values, expected):
```

Comment:
Is the lexicographic order expected? If so maybe add a comment here. If it's a strict API contract it should also be in the docstring.

Reply:
lexicographic order is not expected. Updated with more test to check for unordered cases.